src/app/contexts/ToggleContext/index.jsx

```diff
@@ -1,16 +1,63 @@
-import React, { createContext, useReducer } from 'react';
-import { node } from 'prop-types';
-import toggleReducer from '../../reducers/ToggleReducer';
+import React, { createContext, useEffect, useReducer } from 'react';
+import { node, string } from 'prop-types';
+import webLogger from '#lib/logger.web';
+import { toggleReducer, updateToggles } from './reducer';
 import defaultToggles from '#lib/config/toggles';
+import constructTogglesEndpoint from './utils/constructTogglesEndpoint';
 
+const logger = webLogger();
 const ToggleContext = createContext({});
 
-const ToggleContextProvider = ({ children }) => {
+const ToggleContextProvider = ({ children, service, origin }) => {
+  const simorghToggles = defaultToggles[process.env.SIMORGH_APP_ENV || 'local'];
   const [toggleState, toggleDispatch] = useReducer(
```

Comment:
Do we need to introduce a reducer here? Would useState cover all of our requirements for updating the toggles? As we only have one action type that updates the value of the remote toggle? Could just use useState or the functional useState if that's clearer.

Reply:
> Do we need to introduce a reducer here? Would useState cover all of our requirements for updating the toggles? As we only have one action type that updates the value of the remote toggle? Could just use useState or the functional useState if that's clearer.

yes and no. useState would work just fine, however this is introducing the action reducer pattern which will allow for easier future updates and in my opinion should be used through simorgh when making changes to what is effectively global state.

The dispatching of a single method makes it much easier for any component within the application to cause an update to the toggles context, in the future we may want the service worker to be able issue this update.